File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        d={"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000}
        i=0
        tot=0
        l=True
        while i<len(s)-1:
            if d[s[i]]<d[s[i+1]]:
                tot+=(d[s[i+1]]-d[s[i]])
                if i+3==len(s):
                    tot+=d[s[-1]]
                i+=2
                l=False            
            else:
                tot+=d[s[i]]
                i+=1
                l=True
        if l:
            tot+=d[s[-1]]
        return tot

File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string nums) {
        unordered_map<char,int> m;
        m['I']=1;
        m['V']=5;
        m['X']=10;
        m['L']=50;
        m['C']=100;
        m['D']=500;
        m['M']=1000;
        int ans=0;
        int temp=m[nums[0]];
        for(int i=1;i<nums.size();i++){
            cout<<temp<<" "<<ans<<endl;
            if(m[nums[i]]<=m[nums[i-1]]){
                temp+=m[nums[i]];
            }
            else{
                ans+=m[nums[i]]-temp;
                temp=0;
            }
        }
        ans+=temp;
        return ans;
    }
};